code/go/0chain.net/chaincore/chain/state/state_context.go

```diff
@@ -163,6 +164,22 @@ func (sc *StateContext) AddSignedTransfer(st *state.SignedTransfer) {
 func (sc *StateContext) AddMint(m *state.Mint) error {
 	sc.mutex.Lock()
 	defer sc.mutex.Unlock()
+	data, err := json.Marshal(event.Mint{
+		BlockHash: sc.block.Hash,
+		Amount:    int64(m.Amount),
+	})
+	if err != nil {
+		return fmt.Errorf("marshalling mint: %v", err)
+	}
+	sc.events = append(sc.events, event.Event{
```

Comment:
What we actually want to send to the event database is the total amount minted in a block, not each individual  mint.

I suggest replacing this with emitting a total minted event in [updateState](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/chaincore/chain/state.go#L316). Total mints in for loop and emit afterwards.

Reply:
done